contracts/strategies/VariableRewardsStrategy.sol

```diff
@@ -0,0 +1,304 @@
+        uint256 withdrawAmount = _withdrawFromStakingContract(_amount);
+        uint256 withdrawFee = _calculateWithdrawFee(depositTokenAmount);
+        depositToken.safeTransfer(msg.sender, withdrawAmount.sub(withdrawFee));
+        _burn(msg.sender, _amount);
+        emit Withdraw(msg.sender, depositTokenAmount);
+    }
+
+    function _getWithdrawFeeBips() internal view virtual returns (uint256) {
+        return 0;
+    }
+
+    function _calculateWithdrawFee(uint256 _amount) internal view virtual returns (uint256) {
+        uint256 withdrawFeeBips = _getWithdrawFeeBips();
+        return _amount.mul(withdrawFeeBips).div(_bip());
+    }
+
+    function reinvest() external override onlyEOA {
+        _reinvest();
+    }
+
+    function _convertRewardsIntoWAVAX() private returns (uint256) {
+        uint256 avaxAmount = WAVAX.balanceOf(address(this));
+        uint256 count = rewardCount;
+        for (uint256 i = 0; i < count; i++) {
+            address reward = supportedRewards[i];
+            if (reward == address(WAVAX)) {
+                uint256 balance = address(this).balance;
+                if (balance > 0) {
+                    WAVAX.deposit{value: balance}();
+                    avaxAmount = avaxAmount.add(balance);
+                }
+                continue;
+            }
+            uint256 amount = IERC20(reward).balanceOf(address(this));
+            if (amount > 0) {
+                address swapPair = rewardSwapPairs[reward];
+                if (swapPair > address(0)) {
+                    avaxAmount = avaxAmount.add(DexLibrary.swap(amount, reward, address(rewardToken), IPair(swapPair)));
+                }
+            }
+        }
+        return avaxAmount;
+    }
+
+    /**
+     * @notice Reinvest rewards from staking contract to deposit tokens
+     * @dev Reverts if the expected amount of tokens are not returned from the staking contract
+     */
+    function _reinvest() private {
+        _getRewards();
+        uint256 amount = _convertRewardsIntoWAVAX();
+        require(amount >= MIN_TOKENS_TO_REINVEST, "VariableRewardsStrategy::Reinvest amount too low");
+
+        uint256 devFee = amount.mul(DEV_FEE_BIPS).div(BIPS_DIVISOR);
+        if (devFee > 0) {
+            rewardToken.safeTransfer(devAddr, devFee);
+        }
+
+        uint256 reinvestFee = amount.mul(REINVEST_REWARD_BIPS).div(BIPS_DIVISOR);
+        if (reinvestFee > 0) {
+            rewardToken.safeTransfer(msg.sender, reinvestFee);
+        }
+
+        uint256 depositTokenAmount = _convertRewardTokenToDepositToken(amount.sub(devFee).sub(reinvestFee));
+
+        _stakeDepositTokens(depositTokenAmount);
+        emit Reinvest(totalDeposits(), totalSupply);
+    }
+
+    function _stakeDepositTokens(uint256 _amount) private {
+        require(_amount > 0, "VariableRewardsStrategy::Stake amount too low");
+        _depositToStakingContract(_amount);
+    }
+
+    function _checkReward() internal view returns (Reward[] memory, uint256) {
```

Comment:
this should more closely match the logic of `_convertRewardsIntoWAVAX()`. specifically we miss `address(this).balance`

Reply:
fixed